### DISCUSSION TITLE
More robust Ollama configuration options

### DISCUSSION THREAD

**Discussion by ktuccy** [alagrede/znote-app#87] [Ideas]
This app looks like it will be useful for me. I'd like to integrate Ollama, but the way I've got things setup I run a local server that acts as an Ollama endpoint for all of my various machines and projects.

It would be useful if on the AI settings configuration page you could specify a local network address to use as an API endpoint for Ollama integration.

**Comment by alagrede** [comment]
Hi @ktuccy 👋,

Thank you for your question!

The Ollama plugin installs custom functions along with the necessary NPM packages to facilitate communication with LLMs. By default, the useLocalAI function targets a local Ollama instance, but you can easily customize it or create a new function to specify the parameters you need (see screenshot).

I hope this helps! Let me know if this answers your question or if you'd like me to explore additional solutions to better assist you.

![customFunctions](https://github.com/user-attachments/assets/8b556c69-f256-4cbf-ad72-3830bb1ba2e4)

**Comment by alagrede** [comment]
Hi @ktuccy,  

I’ve updated the Ollama plugin to include a pre-installed function with a configurable `baseURL`. Additionally, the [tutorial](https://recipe.znote.io/recipes/ollama_plugin_tutorial) has been updated to demonstrate how to modify the `baseURL` for your needs.  

I hope this helps! Feel free to reply to this thread or reopen the discussion if you need further adjustments or additional support. 😊

**Comment by michel-briand** [comment]
Good job, Anthony ! After installing the ollama plugin, one must restart Znote (improve the doc to note this).
I play with it and I like it :) !

**Reply by alagrede** [reply]
Thanks a lot Michel! 😊
Good catch. I’ll double-check the behavior. A restart shouldn’t be required in theory, so I’ll investigate and make sure everything behaves as expected (and update the docs if needed).

Really glad to hear you’re playing with it and enjoying it 😎
